pkg/observer/observer.go

```diff
@@ -75,48 +68,56 @@ func (d *dcas) Put(ops []batch.Operation) error {
 		if err != nil {
 			return errors.Wrapf(err, "json marshal for op failed")
 		}
-		_, err = d.getDCASClient().Put(docNs, docColl, bytes)
+		_, err = d.getDCASClient().Put(common.DocNs, common.DocColl, bytes)
 		if err != nil {
 			return errors.Wrapf(err, "dcas put failed")
 		}
 	}
 	return nil
 }
 
-func (d *dcas) getDCASClient() dcasClient {
+func (d *dcas) getDCASClient() client.DCAS {
+	return d.clientProvider.ForChannel(d.channelID)
+}
 
-	d.RLock()
-	dcas := d.dcas
-	d.RUnlock()
+// Observer observes the ledger for new anchor files and updates the document store accordingly
+type Observer struct {
+	cfg          cfg
+	dcasProvider dcasClientProvider
+}
 
-	if dcas != nil {
-		return dcas
+// New returns a new Observer
+func New(cfg cfg) *Observer {
+	dcasProvider := getDCASClientProvider()
+	return &Observer{
+		cfg:          cfg,
+		dcasProvider: dcasProvider,
 	}
-	dcas = getDCAS(d.channelID)
-
-	d.Lock()
-	defer d.Unlock()
-
-	d.dcas = dcas
-	return dcas
 }
 
 // Start starts channel observer routines
-func Start(cfg cfg) error {
-
-	if roles.HasRole(observerRole) {
-
-		logger.Infof("peer is an observer, channels to observe: %s", cfg.GetChannels())
-		for _, channelID := range cfg.GetChannels() {
-			// register to receive Sidetree transactions from blocks
-			n := notifier.New(getBlockPublisher(channelID))
-			dcasVal := &dcas{channelID: channelID}
-			sidetreeobserver.Start(n, dcasVal, dcasVal)
+func (o *Observer) Start() error {
+	for _, channelID := range o.cfg.GetChannels() {
+		if err := o.start(channelID); err != nil {
+			return err
 		}
-
-	} else {
-		logger.Info("peer is not an observer, nothing to do...")
 	}
+	return nil
+}
 
+func (o *Observer) start(channelID string) error {
+	if roles.HasRole(observerRole) {
+		logger.Infof("Starting observer for channel [%s]", channelID)
+		// register to receive Sidetree transactions from blocks
+		n := notifier.New(getBlockPublisher(channelID))
+		dcasVal := newDCAS(channelID, o.dcasProvider)
+		sidetreeobserver.Start(n, dcasVal, dcasVal)
```

Comment:
I'm wondering if Start() returns an error here..

Reply:
No it doesn't.

Reply:
then it's fine